collaboration/extending-kyma-cli.md

```diff
@@ -0,0 +1,346 @@
+
+*Bind to environment variables*
+
+`kyma function bind <function-name> --env [<foo=bar>,..] -n <namespace>`
+`kyma function unbind <function-name> --env [<foo=bar>,..] -n <namespace>`
+
+*List all functions*
+
+`kyma function list -n <namespace>`
+
+> A detailed table view with function name, runtime, age, replicas and state
+
+*Runtimes*
+
+`kyma function get runtimes`
+
+*Logs*
+
+`kyma function logs <function-name> -n <namespace>`
+
+*function Status*
+
+`kyma function status <function-name> -n <namespace>`
+
+Should display status
+
+```bash
+  Name: <function-name>
+  Runtime: <my-runtime>
+  State: Running/Error/Deploying/...
+  Replicas: 1
+  Min-replicas: 1
+  Max-replicas: 2
+  API: <url>
+    Secure: yes/no
+    Actions: GET, POST
+  Bindings:
+    Events: <list of events>
+    Servicebindings: <list of service bindings>
+  labels: foo:bar
+```
+
+`kyma function show <function-name> events`
+`kyma function show <function-name> apis`
+`kyma function show <function-name> service-binding-usages`
+`kyma function show <function-name> labels`
+`kyma function show <function-name> replicas`
+
+#### Typical User flow
+
+This section details a typical development flow for the functions
+
+1. New User creating function from scratch or from existing github project as shown [here](#creation-of-project-locally). This would generate a function project for vscode. User is also provided with readme file to explain the structure of project and also the config.yaml. It should also explain how user can test it locally and also debugging.
+
+2. Once the project is initialized user can deploy the function onto an exisiting kyma cluster using command [here](#commands-that-use-config)
+
+3. User can change the code and test it locally by running a server locally on his workstation. User can make use of make file provided.
+
+4. For debugging purpose user can either run it locally and attach the debuger or for in-cluster debugging he can use the `kyma function debug` command which would use telepresence to redirect the event triggers for example to the local code and he can attach the debugger to debug.
+
+### Service catalog
+
+We should enable kyma cli to be able to configure service catalog too. We should support following commands. The implementation of the commands should be abstracted in a separate library:
+
+`kyma service-instance list`
+
+#### Service Bindings
+
+##### CRUD
+
+`kyma service-binding create <service-binding-name> -n <namespace> --service-instance <service-instance-name>`
+
+`kyma service-binding delete <service-binding-name> -n <namespace>`
+
+`kyma service-binding get <service-binding-name> -n <namespace>`
+
+#### Service Binding Usage
+
+##### CRUD
+
+`kyma service-binding-usage create <sbu-name> -n namespace --service-binding <service-binding-name> --function <function-name>`
+
+`kyma service-binding-usage create <sbu-name> -n namespace --service-binding <service-binding-name> --microservice <microservice-name>`
```

Comment:
So far in the proposal kyma cli was not mentioning microservices. If we support them as well in addition to functions, shouldn't we have a kyma microservice command as well? Or can the function command be used for it somehow?

Reply:
service catalog should be discussed in detail.